lib/bulk-apply-applications.js

```diff
@@ -205,23 +170,25 @@ function bulkApplyApplications (args, callback) {
     function retrieveParentsForUsers (profiles, done) {
       //for each profile, carry out this function
       async.map(profiles, function(profile, cb) {
-        //load the parents of this profile and assign them to the profile
-        seneca.act({role: 'cd-profiles', cmd: 'load_parents_for_user', userId: profile.userId, user: args.user}, function(err, parents) {
-          if (err) return done(err);
-          profile.parents = parents;
+        //if this profile has parents to load
+        if (profile.parents) {
+          //load the parents of this profile and assign them to the profile
+          seneca.act({role: 'cd-profiles', cmd: 'load_parents_for_user', userId: profile.userId, user: args.user}, function(err, parents) {
+            if (err) return done(err);
+            profile.parents = parents;
+            return cb(null, profile);
+          });
+        } else {
           return cb(null, profile);
-        });
+        }
       }, done);
     }
 
     function retrieveEventAndSessionData (profiles, done) {
       profiles = _.uniq(profiles, function (profile) { return profile.id; });
-      seneca.act({role: plugin, cmd: 'getEvent', id: applications[0].eventId}, function (err, event) {
+      seneca.act({role: plugin, cmd: 'searchSessions', query: {id: applications[0].sessionId}}, function (err, sessions) {
         if (err) return done(err);
-        seneca.act({role: plugin, cmd: 'searchSessions', query: {id: applications[0].sessionId}}, function (err, sessions) {
-          if (err) return done(err);
-          return done(null, profiles, event, sessions[0]);
-        });
+        return done(null, profiles, eventData, sessions[0]);
```

Comment:
It's defined at the top and given a value when getEvent is called in loadValidationData(). We defined it like this so that getEvent wouldn't have to be called more than once as it was called three times before.